homeassistant/components/media_player/denonavr.py

```diff
@@ -41,11 +42,49 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Denon platform."""
     import denonavr
 
-    receiver = denonavr.DenonAVR(config.get(CONF_HOST), config.get(CONF_NAME))
-
-    add_devices([DenonDevice(receiver)])
-    _LOGGER.info("Denon receiver at host %s initialized",
-                 config.get(CONF_HOST))
+    # Initialize list with receivers to be started
+    receivers = []
+
+    cache = hass.data.get(KEY_DENON_CACHE)
+    if cache is None:
+        cache = hass.data[KEY_DENON_CACHE] = set()
+
+    # Start assignment of host and name
+    # 1. option: manual setting
+    if config.get(CONF_HOST) is not None:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        # Check if host not in cache, append it and save for later starting
+        if host not in cache:
+            cache.add(host)
+            receivers.append((host, name))
+    # 2a. option: discovery using netdisco
+    if discovery_info is not None:
+        host = discovery_info[0]
+        name = discovery_info[1]
+        # Check if host not in cache, append it and save for later starting
+        if host not in cache:
+            cache.add(host)
+            receivers.append((host, name))
+    # 2b. option: discovery using denonavr library
+    else:
```

Comment:
I would only trigger a discovery if the platform got loaded with no discovery info and no CONF_HOST in config.

Now you will run discovery even if platform got loaded with a host variable.